qualtran/bloqs/state_preparation/state_preparation_alias_sampling.py

```diff
@@ -209,21 +209,31 @@ def qrom_bloq(self) -> QROM:
             (self.alternates_bitsize, self.keep_bitsize),
         )
 
-    def decompose_from_registers(
-        self,
-        *,
-        context: cirq.DecompositionContext,
-        **quregs: NDArray[cirq.Qid],  # type:ignore[type-var]
-    ) -> Iterator[cirq.OP_TREE]:
-        yield PrepareUniformSuperposition(self.n_coeff).on(*quregs['selection'])
+    def build_composite_bloq(self, bb, **soqs):
+        soqs['selection'] = bb.add(
+            PrepareUniformSuperposition(self.n_coeff), target=soqs['selection']
+        )
         if self.mu == 0:
-            return
-        selection, less_than_equal = quregs['selection'], quregs['less_than_equal']
-        sigma_mu, alt, keep = quregs['sigma_mu'], quregs['alt'], quregs['keep']
-        yield cirq.H.on_each(*sigma_mu)
-        yield self.qrom_bloq.on_registers(selection=selection, target0_=alt, target1_=keep)
-        yield LessThanEqual(self.mu, self.mu).on(*keep, *sigma_mu, *less_than_equal)
-        yield CSwap.make_on(ctrl=less_than_equal, x=alt, y=selection)
+            return soqs
+        selection, less_than_equal = soqs['selection'], soqs['less_than_equal']
+        sigma_mu, alt, keep = soqs['sigma_mu'], soqs['alt'], soqs['keep']
```

Comment:
please do this unpacking in the method signature

Reply:
It'll be annoying to create the return dictionary twice when we want to return early. When there are these many parameters, its more convenient to work with the dictionary directly ?

Reply:
oh I see

Reply:
do we need to support the mu = 0 case? what is mu? can you update the docstring to the standard format with an Args: section and a Registers: section. I think the existing Signature section is out-of-date

Reply:
Yes, `self.mu` case needs to be supported. `mu` is a parameter which depends on the precision and can be 0 for when precision is low (eg: probability_epsilon=0.5)

The docstring does seem to be updated and the signature of the bloq hasn't changed since it was written. I'll polish it a bit once I fix the failing tests.

Reply:
what do you mean? there is no args section and the register names don't match the register names

Reply:
updated the docstring